### PR TITLE
Chat plugins: ignore links when determining room FAQ length

### PR DESCRIPTION
**Pull request by Kaiepi**
The current limit's much too short for Bot Development's room FAQ on
bots' nicks and git repositories.

### CODE REVIEW

**Comment by Zarel** [comment]
@bumbadadabum?

**Comment by Kaiepi** [comment]
This could also be useful for other rooms that need to add longer FAQs to explain the topic

**Comment by bumbadadabum** [comment]
But why would you ever need 2500 characters. That's reaaaally big. Why not bump it up slightly. These entries are supposed to be short, look at /faq. If you need 2500 characters you're doing it wrong in my opinion.

**Comment by Zarel** [comment]
image

bumba considers this an unreasonable length for an rfaq entry.

**Comment by bumbadadabum** [comment]
Another approach would be to not count formatting towards the character limit. I also kinda want to implement a limit to the amount of line breaks given the potential shitshow. I don't think 2500 characters of text should ever be needed. Personally I like the "don't make formatting count" option the most if you really need more characters.

**Comment by bumbadadabum** [comment]
image

Even when you're picking something a little dense on text and not a song lyric this seems way overkill to allow in a roomfaq entry. I really prefer going the strip formatting route.

**Comment by Kaiepi** [comment]
Is there a rool to strip formatted text out of a string or would I need to write one myself?

**Comment by Zarel** [comment]
if it's HTML, you can just remove everything between `<` and `>`

if it's PS protocol code, you can probably just write a regex for link syntax.

**Comment by Kaiepi** [comment]
Updated

File: server/chat-plugins/room-faqs.js [hunk 1]
```diff
@@ -47,8 +47,17 @@ const commands = {
 		topic = toId(topic);
 		if (!(topic && rest.length)) return this.parse('/help roomfaq');
 		let text = rest.join(',').trim();
+		// Strip out any potential links. The Perl 6 equivalent to this regex
+		// is included here since it's easier to read:
+		// [
+		// | '[[' [ <-[']']> | ']' <!before ']'> ]* ']]'
+		// | [ [ https? | ftp ] '://' ]?
```
Comment (Zarel) [inline server/chat-plugins/room-faqs.js]: imo regular links shouldn't be stripped, since they appear verbatim anyway

Comment (Kaiepi) [reply server/chat-plugins/room-faqs.js]: So they should be included when checking the length?

Comment (Zarel) [reply server/chat-plugins/room-faqs.js]: Like, the length should be the literal actual length.

`[[this is a test <google.com>]]`

should be treated as

`this is a test`

and

`google.com`

should be treated as

`google.com`

**Comment by Kaiepi** [comment]
Like this?

**Comment by Zarel** [comment]
Looks fine to me aside from the extraneous space. @bumbadadabum?

**Comment by bumbadadabum** [comment]
Seems good to me!

**Comment by Kaiepi** [comment]
Removed the extra space

**Pull request merged**